crates/full-node/sov-db/src/storage_manager/nomt_based/mod.rs

```diff
@@ -295,7 +296,7 @@ where
         let state_overlay = state.into_state_overlay();
 
         let rockbound_snapshot = SnapshotGroup {
-            historical_state: Arc::new(historical_state),
+            historical_state,
```

Comment:
Maybe I am missing something, but why not to have Arcs to be consistent with other change sets, related to previous comment https://github.com/Sovereign-Labs/sovereign-sdk/pull/1430/files#r2253966131